### PR TITLE
Add sysfs-interface-test

### PR DESCRIPTION
**Pull request by vishalbhoj**

### CODE REVIEW

**Comment by mwasilew** [comment]
@vishalbhoj nice job!
I think it might be a good idea to allow users to select which tests to run on their platform. For example the test will always fail on platform without BT and not all boards have BT. Also not all kernels enable BT.

File: automated/linux/peripherals/sysfs-interface-test.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+metadata:
+    format: Lava-Test Test Definition 1.0
+    name: module-tests
```
Comment (nareshkamboju) [inline automated/linux/peripherals/sysfs-interface-test.yaml:3]: please change test
name: sysfs-interface-test

File: automated/linux/peripherals/sysfs-interface-test.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                 USB Gadget Framwork test
+    maintainer:
+        - anders.roxell@linaro.org
+    os:
+        - debian
+        - ubuntu
+        - openembedded
+    scope:
+        - functional
+    devices:
```
Comment (nareshkamboju) [inline automated/linux/peripherals/sysfs-interface-test.yaml:14]: devices:
Please add known good LAVA connected devices which supports any of these test cases.

File: automated/linux/peripherals/sysfs-interface-test.sh [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        echo "Checking connected displays..."
+        xrandr_output=$(xrandr --listmonitors | grep 'Monitors')
+        if [ -n "$xrandr_output" ]; then
+            echo "Connected displays:"
+            xrandr --listmonitors | grep -v 'Monitors'
+        else
+            echo "No displays connected."
+        fi
+    else
+        echo "drm-sysfs-test fail" >> $RESULT_FILE
```
Comment (nareshkamboju) [inline automated/linux/peripherals/sysfs-interface-test.sh:182]: do we have any scope to skip test cases ?
When Kconfigs not enabled / device do not have these peripherals / interfaces we could skip those tests.

**Review by nareshkamboju — CHANGES_REQUESTED** [review]
@vishalbhoj  Thanks taking your time in writing a new test cases.
I have a few minor comments.

**Comment by vishalbhoj** [comment]
@mwasilew Thank you for the feedback. I was just experimenting on the testcase. I'll be closing the PR for now. I'll create a new one with once I have improved the testcase.

File: automated/linux/peripherals/sysfs-interface-test.sh (hunk 3 above)
Comment (vishalbhoj) [reply automated/linux/peripherals/sysfs-interface-test.sh:182]: Thank you @nareshkamboju for the feedback. I'll close the PR for now. I'll open the PR again later once I have improved the implementation.

**Pull request closed without merging**